tpm2/src/util/ski.c

```diff
@@ -0,0 +1,560 @@
+/**
+ * tpm2_kmyth_ski.c:
+ *
+ * C library containing utilities related to the ski format for
+ * Kmyth applications using TPM 2.0.
+ */
+#include "tpm2_kmyth_seal.h"
+#include "tpm2_kmyth_global.h"
+#include "kmyth_cipher.h"
+#include "tpm2_kmyth_misc.h"
+#include "tpm2_kmyth_session.h"
+#include "tpm2_kmyth_io.h"
+#include "tpm2_kmyth_key.h"
+#include "tpm2_pcrManagement.h"
+#include "tpm2_kmyth_object.h"
+#include "tpm2_config_tools.h"
+#include "tpm2_info_tools.h"
+#include "ski.h"
+#include "tpm2_kmyth_mu.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <openssl/rand.h>
+
+//############################################################################
+// tpm2_kmyth_parse_ski_bytes
+//############################################################################
+int tpm2_kmyth_parse_ski_bytes(uint8_t * input, size_t input_length,
```

Comment:
What would you think about making this return the `Ski` structure? It seems weird to me that it gets passed in, but also `free`'d on error within this function.

Reply:
If we made the Ski structure the return value, however, we would have to report errors to the caller differently. Even if we pass the Ski structure as a parameter, though, I think it might be clearer to have the caller free the struct upon the function returning from an error.

Reply:
I have updated this to use a new Ski struct within the function and only modify 'output' when the function executes successfully.